### PR TITLE
[HOTFIX] Post not loaded alternative to text

### LINKED ISSUE #281: [HOTFIX] Post not found page
**Current state**
Post not found results in a completely white page with "Not Loaded" as only text

**Expected state**
A page that handles post not found and other errors related to loading posts

**TO DO**
- [ ] Create not found page
- [ ] Send users to not found page on post not found
- [ ] Test if other errors should be handled as well

**Branch**
hotfix-not-loaded-alternative-error

### PR DESCRIPTION
**Pull request by AjobK**
closes #281 

<br />

 **What is it supposed to do**  
Better loading of post instead of 'Not loaded' text. Added a loader icon when a post is loading. Also gives an error when post is not found instead of a permanent text indicating that something is loading.

<br />

 **How can we test your branch**  
Go to an invalid post URL like `/post/blahblah`. You should see a loader for a very short amount of time and then an error page indicating that the post does not exist.
<br />

 **Additional notes**  
None

<br />

**1. General checklist**
- [x] a. Tested happy flow.
- [x] b. Tested unhappy flow.
- [x] c. No unexpected exceptions.
- [x] d. No code with high complexity. (Big O)
- [x] e. EOF newline.
- [x] f. No linter warnings or errors.

<br />

**2. If relevant, front-end checklist**
- [x] a. Responsive. (Desktop all the way to mobile 320px)
- [x] b. Using SCSS variables where relevant.
- [x] c. Relevant class names
- [x] d. No modified global SCSS properties, except if necessary and fully tested the impact.
- [x] e. Checked [caniuse](https://caniuse.com) for more modern CSS properties.

<br />

**3. If relevant, back-end checklist**
- [x] a. No unhandled promise rejections.
- [x] b. Seeders still functional.
- [x] c. No primary-key violations.
- [x] d. Normalization, at least [3NF](https://en.wikipedia.org/wiki/Database_normalization).
- [x] e. Used guards (middleware) where necessary and tested if route is guarded.

<br />

**4. If relevant, test these browsers**
- [ ] a. Chrome
- [ ] b. Firefox
- [ ] c. Opera
- [ ] d. Edge
- [ ] e. Safari

---

**PR Rules**  
*1. Add relevant labels.*  
*2. Select at least two reviewers.*  
*3. Assign all reviewers allowed to merge the branch, including yourself.*  
*4. If relevant to the sprint, put related issue in 'Awaiting Approval' column on project board.*  
*5. Add to the corresponding [milestone](https://github.com/AjobK/Seaqull/milestones).*

<br />

**Reviewer info**  
*1. You can use aliases to refer to specific PR checklist items. For example 'Hey @JohnDoe, for this line please check 3.a and 1.d.'*  
*2. Where possible, please use the suggestions feature in github so the author has a better idea of the solution you had in mind.*   

<br />

> This form should be filled in by the author of this PR.

### CODE REVIEW

File: backend/app/controllers/postController.ts [hunk 1]
```diff
@@ -83,6 +83,9 @@ class PostController {
       decodedId = -1
     }
 
+    if (!foundPost)
+      return res.status(404).json({ 'message': 'No post found on that path' })
+
```
Comment (danielordaan) [inline backend/app/controllers/postController.ts]: I'd put this just under line 73 (just after you retrieve the potential post from the dao).

File: frontend/src/pages/post/index.js [hunk 2]
```diff
@@ -98,6 +98,23 @@ class Post extends App {
...
+        if (e?.response?.data?.message) {
+          message = e.response.data.message
+        } else if (e?.response?.statusText) {
+          message = e.response.statusText
+        }
+
+        this.props.history.push({
+          pathname: '/error',
+          state: {
+            title: e.response ? e.response.status : name,
```
Comment (danielordaan) [inline frontend/src/pages/post/index.js]: I think renaming 'e' to 'error' makes the code more readable overall.

File: frontend/src/pages/post/index.js [hunk 3]
```diff
@@ -272,7 +290,7 @@ class Post extends App {
             </ReactTooltip>
           </div>
         </Section>
-        { !this.props.new && <CommentSection/> }
+        { !this.props.new && <CommentSection isPostOwner={ this.state.isOwner } /> }
```
Comment (danielordaan) [inline frontend/src/pages/post/index.js]: What is the purpose of adding isPostOwner to the comment section in your issue? I remember doing this in my own issue so I'm wondering what this is doing here

**Review by danielordaan — CHANGES_REQUESTED** [review]
Tested, nice work! Some comments below

File: frontend/src/pages/post/index.js (hunk 3 above)
Comment (AjobK) [reply frontend/src/pages/post/index.js]: You and me both, lol

**Review by danielordaan — APPROVED** [review]
Good work! Approved

**Pull request merged**